src/kademlia/dht_storage.cpp

```diff
@@ -268,26 +267,12 @@ namespace
 			torrent_entry* v;
 			if (ti == m_map.end())
 			{
-				// we don't have this torrent, add it
-				// do we need to remove another one first?
-				if (!m_map.empty() && int(m_map.size()) >= m_settings.max_torrents)
+				if (int(m_map.size()) >= m_settings.max_torrents)
```

Comment:
It doesn't seem worth it to prioritize info hashes by distance. Unlike the number of peers per info hash, the distribution of info hashes themselves is of course uniform so the number of info hashes per node shouldn't vary by much. Whatever form the resource limit takes, it shouldn't be set so low as to cause a node to reject legit announces for a new info hash. Only an attacker should be getting announces for new info hashes rejected.